arq/worker.py

```diff
@@ -280,6 +278,28 @@ def run(self) -> None:
                 queued_jobs = await self.pool.zcard(self.queue_name)
                 if queued_jobs == 0:
                     return
+            async with self.sem:  # don't bother with zrangebyscore until we have "space" to run the jobs
```

Comment:
I don't understand why we're now doing `zrangebyscore` in two different places.

Is this really required?

Reply:
@samuelcolvin This was introduced by mistake, it's absolutely not required. The only thing needed is what is inside `_poll_iteration`. This has been fixed now.